badge_server/test_badges.py

```diff
@@ -135,74 +255,106 @@ def setUp(self):
         self.addCleanup(self._finder_patch.stop)
         self._pkg_list_patch.start()
         self.addCleanup(self._pkg_list_patch.stop)
+        self._whitelist_urls_patch.start()
+        self.addCleanup(self._whitelist_urls_patch.stop)
 
-    def _get_image(self, package):
+    def get_image_json(self, package):
+        """Return the calculated badge data for a package as a dict."""
         return self.client.get(
-            '/one_badge_image', query_string={'package': package})
-
-    def _get_image_json(self, package):
-        return self._get_image(package).get_json()
-
-    @unittest.skip
-    def test_pypi_success(self):
-        self.fake_store.save_compatibility_statuses([
-            GOOGLE_API_CORE_RECENT_SUCCESS_2,
-            GOOGLE_API_CORE_RECENT_SUCCESS_3,
-            APACHE_BEAM_GOOGLE_API_CORE_RECENT_SUCCESS_2,
-            GOOGLE_API_CORE_GOOGLE_API_PYTHON_CLIENT_RECENT_SUCCESS_2,
-            GOOGLE_API_CORE_GOOGLE_API_PYTHON_CLIENT_RECENT_SUCCESS_3,
-            GOOGLE_API_PYTHON_CLIENT_TENSORFLOW_RECENT_SUCCESS_3,
-        ])
-        json_response = self._get_image_json('google-api-core')
+            '/one_badge_image', query_string={
+                'package': package
+            }).get_json()
+
+    def assertLinkUrl(self, package, actual_url):
+        """Assert that the link for the badge image is correct for a package."""
+        o = urllib.parse.urlparse(actual_url)
```

Comment:
What does `o` mean?

Reply:
It was the example used in the docs...I fixed it.